langchain/src/vectorstores/cassandra.ts

```diff
@@ -215,54 +280,172 @@ export class CassandraStore extends VectorStore {
    * @returns Promise that resolves when the database has been initialized.
    */
   private async initialize(): Promise<void> {
-    await this.client.execute(`CREATE TABLE IF NOT EXISTS ${this.keyspace}.${
-      this.table
-    } (
-      ${this.primaryKey.name} ${this.primaryKey.type} PRIMARY KEY,
-      text TEXT,
+    let cql = "";
+    cql = `CREATE TABLE IF NOT EXISTS ${this.keyspace}.${this.table} (
+      ${this.primaryKey.map((col) => `${col.name} ${col.type}`).join(", ")}
+      , text TEXT
       ${
         this.metadataColumns.length > 0
-          ? this.metadataColumns.map((col) => `${col.name} ${col.type},`)
+          ? ", " +
+            this.metadataColumns
+              .map((col) => `${col.name} ${col.type}`)
+              .join(", ")
           : ""
       }
-      vector VECTOR<FLOAT, ${this.dimensions}>
-    );`);
-
-    await this.client
-      .execute(`CREATE CUSTOM INDEX IF NOT EXISTS idx_vector_${this.table}
-  ON ${this.keyspace}.${this.table}(vector) USING 'StorageAttachedIndex';`);
+      , vector VECTOR<FLOAT, ${this.dimensions}>
+      , ${this.buildPrimaryKey(this.primaryKey)}
+    ) ${this.withClause ? `WITH ${this.withClause}` : ""};`;
+
+    await this.client.execute(cql);
+
+    this.selectColumns = `${this.primaryKey
+      .map((col) => `${col.name}`)
+      .join(", ")}
+                          ${
+                            this.metadataColumns.length > 0
+                              ? ", " +
+                                this.metadataColumns
+                                  .map((col) => `${col.name}`)
+                                  .join(", ")
+                              : ""
+                          }`;
+
+    cql = `CREATE CUSTOM INDEX IF NOT EXISTS idx_vector_${this.table}
```

Comment:
Any injection risk?

Reply:
none added - this is part of the vector store "create" process. The PR does resolve injection risks though, as previously the values were being used as part of the literal string, and the update uses bind variables instead.